### ISSUE TITLE
mix phoenix.server / SASS fails on Windows

### ISSUE THREAD

**Issue by lstor** [phoenixframework/phoenix#794]
As noted [on the mailing list](https://groups.google.com/forum/#!topic/phoenix-talk/FB4DrpeN1sc), I can't get Phoenix to work on Windows after repeated efforts.

After digging around, it seems that `events.js` is used by SASS. I have been unable to find a work-around myself so far, but it is probably because I'm not familiar with Phoenix and how things fit together. Any help with resolving the error or finding a workaround to produce CSS and get on with the Phoenix tutorial would be greatly appreciated.

I know this probably isn't an error in Phoenix proper, but since a fresh project in a fresh installation fails when following the tutorial, *something* is definitely wrong =)

**Comment by scrogson** [comment]
The `sass-brunch` package uses `libsass` by default but can be configured to use `ruby` if desired. I think you just need to ensure that you have `libsass` installed.

Here's a couple of (hopefully) helpful links: 

https://github.com/brunch/sass-brunch
https://github.com/sass/libsass/wiki/Building-on-Windows

**Comment by scrogson** [comment]
https://github.com/phoenixframework/phoenix/issues/794#issuecomment-95289125

**Comment by lstor** [comment]
Checking it out, thanks! I'll report back on how it goes.

**Comment by lstor** [comment]
I didn't have any luck with using the ruby builder instead, but I'm wondering if perhaps I found another clue as to what is wrong.

I tried regenerating the Phoenix project, and then I noticed this output from npm:

```
npm WARN optional dep failed, continuing fsevents@0.3.5
```

Combined with the stack trace, my hypothesis is that what's going on is that sass is watching for changed files, and then something fails because the [`fsevents`](https://www.npmjs.com/package/fsevents) package isn't available on Windows.

**Comment by chrismccord** [comment]
So phoenix is actually working just fine, it's just your asset compilation through brunch that's failing. In the meantime before you find a fix, you can use phoenix without brunch by passing the `--no-brunch` flag to `phoenix.new`. We had a handful of people at the Phoenix workshop today and brunch was working... mostlyish. One person had to run `brunch watch` directly on cmd.exe and remove the watcher.

**Comment by josevalim** [comment]
Can we leave this open? We should ensure brunch works out of the box regardless of the OS they are using.

**Comment by josevalim** [comment]
If we find the person that had issues on windows during the workshop, I
would love to debug it and find out what was wrong.

-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Lead Developer*

**Comment by lstor** [comment]
Alright, I've figured it out now =) @chrismccord's comment about running `brunch watch` led me to suspect that maybe the problem was with running brunch. Turns out `brunch` wasn't installed, and when I tried to install it with npm, it failed. I found out that my node and/or npm installations were too old, which meant that brunch didn't install properly.

I simply reinstalled node and npm, and now everything is working properly. Yay!
